egs/wsj/s5/utils/lang/bpe/prepend_words.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# This script, prepend '|' to every words in the transcript to mark
+# the beginning of the words for finding the initial-space of every word
+# after decoding.
+
+import sys, io
+
+infile = io.TextIOWrapper(sys.stdin.buffer, encoding='utf-8')
```

Comment:
For anything that doesn't need to split inside words (like this script) I prefer the latin-1 encoding; that will also work on data like GBK that's not utf-8 but which is compatible with ASCII.